packages/image_picker/tizen/src/image_picker_tizen_plugin.cc

```diff
@@ -53,176 +66,119 @@ class ImagePickerTizenPlugin : public flutter::Plugin {
-              privilege,
-              [](ppm_call_cause_e cause, ppm_request_result_e result,
-                 const char *privilege, void *data) -> void {
-                auto *plugin = (ImagePickerTizenPlugin *)data;
-                assert(plugin);
-
-                if (cause == PRIVACY_PRIVILEGE_MANAGER_CALL_CAUSE_ERROR) {
-                  LOG_ERROR("ppm_request_permission error! [%d]", result);
-                  plugin->SendResultWithError("Permission denied");
-                  return;
-                }
-                if (result !=
-                    PRIVACY_PRIVILEGE_MANAGER_REQUEST_RESULT_ALLOW_FOREVER) {
-                  LOG_ERROR("ppm_request_permission deny! [%d]", result);
-                  plugin->SendResultWithError("Permission denied");
-                  return;
-                }
-
-                plugin->PickContent();
-              },
-              this);
-          if (ret != PRIVACY_PRIVILEGE_MANAGER_ERROR_NONE) {
-            LOG_ERROR("ppm_request_permission fail! [%d]", ret);
-            break;
-          }
-          return;
-        default:
-          LOG_ERROR("ppm_check_permission deny!");
-          break;
-      }
+  bool CheckPermission() {
+    PermissionManager manager;
+    PermissionResult result =
+        manager.RequestPermssion("http://tizen.org/privilege/mediastorage");
+
+    if (result == PermissionResult::kDenyForever ||
+        result == PermissionResult::kDenyOnce) {
+      SendErrorResult("Permission denied", "Permission denied by user.");
+      return false;
+    } else if (result == PermissionResult::kError) {
+      SendErrorResult("Operation failed", "Failed to request permission.");
+      return false;
     }
-    SendResultWithError("Permission denied");
-#else
-    PickContent();
-#endif
+    return true;
   }
 
-  void PickContent() {
+  void PickContent(std::string mime_type) {
+    if (!CheckPermission()) {
+      return;
+    }
+
     app_control_h handle = nullptr;
-#define RET_IF_ERROR(ret)                                             \
-  if (ret != APP_CONTROL_ERROR_NONE) {                                \
-    SendResultWithError(std::to_string(ret), get_error_message(ret)); \
-    if (handle) {                                                     \
-      app_control_destroy(handle);                                    \
-    }                                                                 \
-    return;                                                           \
+#define RETURN_IF_ERROR(ret)                                      \
+  if (ret != APP_CONTROL_ERROR_NONE) {                            \
+    SendErrorResult(std::to_string(ret), get_error_message(ret)); \
+    if (handle) {                                                 \
+      app_control_destroy(handle);                                \
+    }                                                             \
+    return;                                                       \
   }
```

Comment:
I also considered this but I thought it would add too much complexity. (Notice that the API `app_control_get_extra_data_array` is used again in the reply callback.)

I'll check again, but I'm not sure the change will do any better.

Reply:
Thank you for your answer. 
This PR is already excellent and contain a lot of changes, so I think we can think about it again next time :)

Reply:
Alright. Thanks for the suggestion! I'll keep this principle in mind in later commits.